TMS/CourseCreation: changed Step interface...

klees opened the pull request:
... and added a class for a request type.

https://jira.cat.internal/browse/TMS-616

klees committed:
TMS/CourseCreation: added type for Request

klees committed:
TMS/CourseCreation: added request builder to steps

shecken commented on src/TMS/CourseCreation/Request.php:
```diff
@@ -0,0 +1,104 @@
+	 */
+	protected $finished_ts;
+
+	public function __construct($id, $user_id, $session_id, $crs_obj_id, \DateTime $requested_ts, \DateTime $finished_ts = null) {
```
Hier fehlen noch die Docstrings für die Parameter die nicht gehinted sind.

shecken commented on src/TMS/CourseCreation/Wizard.php:
```diff
@@ -37,19 +37,24 @@ class Wizard implements \ILIAS\TMS\Wizard\Wizard {
 	 * @param	int	$timestamp				timestamp the process was started
 	 */
-	public function __construct($dic, $user_id, $crs_ref_id, $timestamp) {
+	public function __construct($dic, RequestBuilder $request_builder, $user_id, $crs_ref_id, $timestamp) {
```
Docstring passt nicht zum constructor

shecken replied:
Das passt immer noch nicht. Im Docstring sind parameter die es gar nicht mehr  gibt.

klees committed:
TMS/CourseCreation: added Docstring

klees committed:
TMS/CourseCreation: fixed Docstring

klees commented:
Done!

klees committed:
TMS/CourseCreation: ref_id works as well as obj_id

klees committed:
TMS\CourseCreation: turned RequestBuilder into a richer interface.

In this way it can be implemented in some database class. This seems
to be a good idea, since the builder actually needs to create ids, which
is what the database does in our standard implementations.

klees committed:
TMS/CourseCreation: use new builder in wizard
